### ISSUE TITLE
Channel blocking feature

### ISSUE THREAD

**Issue by okan35** [TeamNewPipe/NewPipe#1815]
Will there be a channel blocking feature because this feature is really important in my opinion so we won't get any videos from the channels we don't want.

**Comment by TobiGr** [comment]
Well, that's a difficult question. A feature like this has always two sides.
One one hand, this would allow to block these spam YouTuber's who do nothing else then click-bait and and so on. IMO a good reason to have a feature like this.
On the other hand, this is pretty dangerous. As soon as I block channels because of a different view or opinion, we create something NewPipe tries to avoid: the well know filter-bubble. Diversity is important in all parts of our live. 
So I'd say, in case someone implements this, there must always be a note that a video has been removed and the ability to show it, must be given. Furthermore, a warning about the risks should be shown when the feature is enabled in the settings.
But IMO we should not implement this because this allows to suppress different opinions.
@TheAssassin @theScrabi @All what do you think?

**Comment by lionirdeadman** [comment]
Any feature can be abused if given to the wrong hands. I think we should allow this feature. 

Let's assume the best out of people instead of the worse. It's not productive otherwise in my opinion.

**Comment by okan35** [comment]
I think people should be free to watch what they want and what they don't want. For example I against those unnecessary make up videos, I don't like them so I block those channels or I am not into football so I don't want to see anything related to football and so on. I implemented the base of this feature in Skytube and I love it. It is definitely a necessary feature. In my opinion this feature is definitely not about blocking ideas that we don't like, it would be like that if I said "let's block anything from Germany".

**Comment by TheAssassin** [comment]
I don't see any reason not to allow blocking channels. It's a lot harder to create "filter bubbles" by allowing users to manually block channels than any sort of automated filtering on server side. I don't see any reason not to do it, and I'm not quite sure what @TobiGr is trying to describe...

**Comment by vynmera** [comment]
@TobiGr I think OP is talking about a local block, as in "I don't like seeing videos from (x), I will block them" instead of some global block as in "NewPipe devs don't like (x) so it's blocked"

**Comment by okan35** [comment]
Of course I definitely didn't mean to block server wide. I wanted it to be done locally for personal preferences.

**Comment by TobiGr** [comment]
> Of course I definitely didn't mean to block server wide. I wanted it to be done locally for personal preferences.

Yeah, I mixed this up. My fault. Sure we can do this. But maybe we should add an option to show hidden videos.

**Comment by okan35** [comment]
Yeah that is the way to go. It can a blocking preference with unblocking option.

**Comment by speednroll** [comment]
this is related
https://github.com/TeamNewPipe/NewPipe/issues/1243

**Comment by Stypox** [comment]
What about just hiding the thumbnail of videos from blocked channels? Most of the times the title is not so "clickbaity", while the thumbnail is. This would not close users in a bubble, rather it would prevent them from getting distracted and lose their time.

**Comment by GoatPlushie** [comment]
@Stypox You still have so scroll past them / read the titles though. While thumbnail hiding would be nice, it wouldn't save as much time and distraction as just scrubbing them completely.

**Comment by Ecksters** [comment]
Would love to see a whitelist version of this, great for setting up a tablet for kids to use without them getting to all the garbage on YT.

**Comment by TobiGr** [comment]
@drji3m came up with the idea of creating a blacklist for videos, too (see #2596). I think this would be a nice addition, especially when you listen to music using the background player with autoplay enabled.

**Comment by drji3m** [comment]
I main want it as a way to "weed the garden" for personal use. You tube has a high percentage of "vanity videos" published by folk who don't care if its useless or annoying. I typically search for instructional "how to " videos and am totally frustrated when the search is full of useless videos that I've already rejected. Also I want to remove videos I've already watched but want to work through a large number to watch (e.g. TV series)

BTW, I can't figure out how "up next" videos are chosen. They are usually completely unrelated.

**Comment by drji3m** [comment]
Let's not call it a personal blacklist. Let's just call it "Been there, seen this" so we can search for stuff we haven't watched. Just a negative playlist.

**Comment by hamiltonrn** [comment]
**Channel Block** feature is much needed, I don't know in other countries, but in Brazil, Trending video list is full of channels for kids and teens, not to mention click-bait videos, I would love to block them. That's the only reason I still use SkyTube. **Please implement Channel Block!**

**Comment by drji3m** [comment]
My problems are multiple. I need some way of marking/blocking videos I have
watched since most TV series episodes are not sorted. Also need to somehow
block videos by idiots who add nothing to 5he universe. Sometimes its a
channel block but often it's single videos.

I don't know if search results are filtered by Google ( I would hope not
since Google thinks I'm insane).

**Comment by MalakymR** [comment]
> I main want it as a way to "weed the garden" for personal use. You tube has a high percentage of "vanity videos" published by folk who don't care if its useless or annoying. I typically search for instructional "how to " videos and am totally frustrated when the search is full of useless videos that I've already rejected. Also I want to remove videos I've already watched but want to work through a large number to watch (e.g. TV series)
> 
> BTW, I can't figure out how "up next" videos are chosen. They are usually completely unrelated.

YT also adds your rejections into their data harvesting.

> Would love to see a whitelist version of this, great for setting up a tablet for kids to use without them getting to all the garbage on YT.

This would make it better than YoutubeKids as that requires you to have an account/sign in - more tracking (of kids) for Youtube/Google.

**Comment by selurvedu** [comment]
@drji3m Google's algorithms choose the next video for you. NewPipe has nothing to do with that. It could, though, switch to another video from suggestions if "Up next" is blacklisted.

**Comment by MalakymR** [comment]
Bounty posted https://www.bountysource.com/issues/64727085-feature-request-blacklist-videos-and-channels

**Comment by ChunkyProgrammer** [comment]
Should this include hiding comments made by blocked channels/containing blocked keywords or should I make a seperate issue for that?

**Comment by madasadad** [comment]
can anyone explain me how to block videos from a particular channel? if it is not added please take this issue seriously

**Comment by drji3m** [comment]
Ive neen asking for this for years. It shouldnt be hard since the mechanism
for favorite lists exists.

Is there some reason why this wont be implemented?

**Comment by 32423d** [comment]
The functionality to block channels, videos, and keywords is already present in a browser extension for desktop browsers called BlockTube (Github Repository link: https://github.com/amitbl/blocktube). It would be nice if the Newpipe developers were to implement the functionality from this browser extension into the Newpipe app or at least some sort of way to import the block list that can already be exported from BlockTube into Newpipe if Newpipe was to have the ability to block channels, videos, and keywords in the future.

**Comment by newpipemember** [comment]
### **Please Try to add Block_channel feature in Next Update**

**Comment by fhteagle** [comment]
+1 For per-channel hiding from trending, search results, etc.

The YT algorithm stubbornly insists on putting some very obviously click bait low quality journalism channels in my results, channels that are nothing but one person's rants, etc. I have not clicked on any of these channels that I have in mind in a year or more, but because they are covering the same topics I am researching, YT keeps trying to push these videos my way. The real fix is on the YT algorithm side, but since we can't push that rope, allowing users to hide specific channels is the next best thing.

**Comment by Ironbull200** [comment]
I would really appreciate a feature to filter search results by keywords/regular expressions as well, but for different reasons as were already discussed  above:

As you all know, there are some very porn-ish types videos available on YouTube (e.g. JOI videos), which are very problematic for kids and people, who suffer from an addiction to pornography (which are quite many; e.g. the subreddit r/pornfree has almost 260.000 members). It would be incredibly benefitial to have the ability to make a local list of keywords, which get blacklisted. So if the search term or video title contains one of the blacklisted keywords, the video should not show up in the search results and/or at least the video should not play. 

The existing age restriction feature and Youtube's "restricted mode" feature does not do this issue justice for two reasons: 
1. Both can easily be bypassed by simply disabling said feature by the click of a button (a feature to lock these settings could help). Relying on the users willpower to simply not search for such videos doesn't take into account the compulsive nature of such unwanted behaviors. 
2. Both features filter over- and underblock quite heavily at the same time. Many videos get blocked, which shouldn't get blocked and many video do not get blocked, which should get blocked.

A keyword-based/regular expression based filter feature would allow each user to tailor the search results for ones own needs. Other application fields could be people with eating disorders, people with self-harming behaviors or other compulsive behaviors, which hardly get any recognition by YouTube or other services including Newpipe. On a desktop PC there are already multiple browser extensions, which allow such filtering based on regular expressions (e.g. LeechBlock or BlockSite), which are widely popular in said communities. 

Since the issue of bypassing the feature  in question would still be present even if a keyword-based filter option would exist, it should also be accompanied by the possibility to lock the feature after setup by a PIN, Password or time-delay (latter is a core feature of blocking software "Pluckeye", which is super popular among self-regulating members of said communities as well). Otherwise, a keyword-filter feature would still be almost entirely useless.

In regards to the filter bubble problem: Yes, such a feature could potentially lead to the creation of filter bubbles. However, I think we should trust the user to be sensible which content gets blocked out or not. Also, I hope I could make clear that such a feature would also benefit many people immensly, which - I find - is enough of a reason to accept the risk of filter bubbles.

I really hope this feature gets taken into serious consideration, as it would be immensly helpful! Thanks.

**Comment by opusforlife2** [comment]
https://f-droid.org/en/packages/su.sadrobot.yashlang/

**Comment by caveman1973** [comment]
Any update on this feature that was requested in 2018?

A simple blocking of channels would be enough.

If filters could be implemented, great. But it's not nearly as useful as being able to block channels. 

It doesn't look like something that would take that many years to implement, specially with a bounty.

**Comment by Sepero** [comment]
> Would love to see a whitelist version of this, great for setting up a tablet for kids to use without them getting to all the garbage on YT.

I love this idea

**Comment by mojoaxel** [comment]
> Bounty posted https://www.bountysource.com/issues/64727085-feature-request-blacklist-videos-and-channels

I just increased the bounty by 100$ for 3 month. Come join, and let this get implemented! 🤑

**Comment by IshanChhangani** [comment]
This functionality would be great to have.

**Comment by QuAzI** [comment]
Still waiting for this feature. As example YT player on child tablet and even if you use only subscription list under videos regularly shown some trash. Sometimes even with different language.

**Comment by newpipemember** [comment]
Waiting badly for the feature

**Comment by Alexander-Wilms** [comment]
It would be nice if the feature supported regex and one could im- and export blacklists, perhaps in the same format that [BlockTube](https://addons.mozilla.org/en-US/firefox/addon/blocktube/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search) uses.

Format of video title and channel name blacklist:
```
// all-caps
/[A-Z ]15+/
than you think
than you thought
```

Format of video ID blacklist:
```
// Blocked by context menu (Aussies Try Each Other's Expensive Work Lunches)
AF5e3M7nRp4
```

Format of channel ID blacklist:
```
// Blocked by context menu (Quantum Tech HD)
UC4Tklxku1yPcRIH0VVCKoeA

// Blocked by context menu (Marques Brownlee)
UCBJycsmduvYEL83R_U4JriQ
```

**Comment by IshanChhangani** [comment]
As many already pointed out, the [BlockTube](https://addons.mozilla.org/en-US/firefox/addon/blocktube/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search) extension, there is also the [Channel Block Extension](https://chrome.google.com/webstore/detail/channel-blocker/nfkmalbckemmklibjddenhnofgnfcdfp) which more or less does the same work.  It would be great to have this functionality.

**Comment by eklenin** [comment]
Content blacklisting in last days is becoming more of necessity rather than a feature.

Modern people's attention time became a product, so now there is a lot of competition around it. It's not about search bubbles anymore, even if you are not surfing but just trying to find specific content - you will basically get spammed by all kind of irrelevant stuff, that make finding things extremely difficult.

I see things becoming so bad in future, that even blacklisting will be not enough anymore, but at least for some time it'll help.

**Comment by drji3m** [comment]
I've been trying fir years to get this added. It's especially needed for
instructive videos make by those afraid to speak, those who hold a wrench
in one hand and a camera in the other, general unboxing video from folks
who can't open the box and have no clue what they bought. This kind of
clutter make it difficult to find the 5% of truly useful video.

Seems like it would be just like a bookmark list that is used to filter the
search results. Or even the addition of a thumbs up/down flag to the dingle
bookmark list.

**Comment by valtoree** [comment]
just realized I need this and someone said that skytube has it. so please 🙏 implement. ty dev/s for ur hard work ♥️

**Comment by newpipemember** [comment]
Years have been passed, no news has not come yet, cannot uou implement this?

**Comment by offbeat-stuff** [comment]
I personally want to be able to block single videos from automatically joining queue. Sometimes i am looking normal videos and I have to remove the same video again and again from my queue.

As for the argument of filter-bubble I dont wanna hide the video. It can show up. I just wanna be able to shorten the repeated task of removing a video from queue.

**Comment by ryanbarillos** [comment]
On a similar note, we have a FOSS extension that can do this on PC browsers called [BlockTube](https://github.com/amitbl/blocktube): It can block specific videos & block specific YT channels (using channel ID's) & any of their future videos from displaying on the feed.

Currently this is built on JavaScript. But if the implementation can be converted to code for use on NewPipe, this feature should find its way into a future release of this app.

**Comment by drji3m** [comment]
Great. Make Utube seem overpressuring

**Comment by offbeat-stuff** [comment]
sponsorblock fork of newpipe seems to implement this feature.

It is now called tabular.

**Comment by IshanChhangani** [comment]
> sponsorblock fork of newpipe seems to implement this feature.
> 
> It is now called tabular.

@offbeat-stuff can you please share some link for "tabular" where I can check the code. Or any tutorial on how to use this.
I think by 
> sponsorblock fork of newpipe seems to implement this feature.
you mean https://github.com/polymorphicshade/NewPipe.

**Comment by offbeat-stuff** [comment]
@IshanChhangani https://github.com/polymorphicshade/NewPipe/discussions/260

**Comment by RA1-Asur** [comment]
So I have tried different Apps/Youtube clients available on the internet...And yes some of them have video filtering feature :-
1) Clipious , best video filtering imo...but sometimes it buffers very much.
2) Skytube, tried a good app...but less options for filtering 
3) Freetube, in haven't tried it . 
1)
![IMG_20240328_105847](https://github.com/TeamNewPipe/NewPipe/assets/123820449/98275de7-acc8-47b0-a951-003ba31b7345)
2)
![IMG_20240328_105917](https://github.com/TeamNewPipe/NewPipe/assets/123820449/52b16f17-39de-441b-a859-6a394b68a4c2)
3)
![IMG_20240328_110049](https://github.com/TeamNewPipe/NewPipe/assets/123820449/e3f6ca36-eb8b-44fd-86c3-a5bd49870e3c)

And Yes the sponsorblock version of Newpipe by polymorphshade(sorry) is yet to implement video filtering....there was video filtering at some point...but the new app doesn't have this option .
![IMG_20240328_110844](https://github.com/TeamNewPipe/NewPipe/assets/123820449/79a8d5e9-e457-4e52-a305-46892f052fbc)
...
....
....
Basically I'm a student..So the must things I need are
1) Skip Silence 
2) And Video Filtering 
3) I don't care about sponsorblock or RYDislikes
....

...
So if anyone knows any app....I don't care about privacy...Can suggest me any app....I don't have a PC  :⁠-⁠(

**Comment by seanking2919** [comment]
I really hope this happens. I cannot tell you how many times I encounter clickbait videos from political grifters when they're not at all related to what I'm watching.

**Comment by Oshizlgzngahr** [comment]
👍👍👍 Please implement this change. I cannot upvote it enough.

Speaking from experience w/ FreeTube on desktop  it cannot be understated how useful of a feature this is in removing both online junk/clutter from your feed and trending as well as just helping to get better results when trying to find YT content in the app.

Use Cases:
- Trying to look up trending content like movie/game trailers and being bombarded with fake fan trailer clickbait and/or repost channels.
-Trying to find new videos/creators in a genre of content (i.e. video essays) but top results are dominated by videos/channels you aren't interested or already have seen/subscribed to.
- Most obvs. you just dont want to see content from a certain channel/creator for whatever reason.

**Comment by novastar88** [comment]
I really miss this feature. Something like [BlockTube](https://github.com/amitbl/blocktube) (Firefox extension) would be awesome.

**Comment by offbeat-stuff** [comment]
For those still looking.
There is https://github.com/lamarios/clipious

Has good search filter

As for blocking
can block channel names,video titles,video lengths
based on schedule
either mark as obfuscate or hide completely

**Comment by solarwanderer00** [comment]
Lately at least 20% of my recommended tab is conspiracy theorists / political grifters despite the fact that their content has nothing to do with the video I'm watching.
Without a way to block these channels I might as well give up on youtube since the crazies are turning it into yet another nazi bar.

**Comment by novastar88** [comment]
@solarwanderer00 
"grifters", "nazi", "crazies"... keep you neo-marxist ideology out of here, there are better places to express your political frustration.

**Comment by MurzNN** [comment]
I found another app that gives the Whitelist/Blackist functionality - Grayjay https://grayjay.app/, so who needs this feature - until this is implemented in NewPipe, you can try this app.

**Comment by drji3m** [comment]
Thanks

**Comment by apogorzelska** [comment]
Having a keyword blacklist would be excellent, I hope this gets implemented

**Comment by hero23550** [comment]
waiting for this feature...

**Comment by offbeat-stuff** [comment]
Its implemented in pipepipe

**Comment by mstudio005** [comment]
I made a PR for this #12787 in November, reviewer asked me to target refactor branch instead of dev and also had some changes. I was ready to fix them but got confused — is this feature still needed in refactor or is someone already working on it there? If needed I'll port it and fix the changes.

**Comment by absurdlylongusername** [comment]
@mstudio005 All non-negligible features must target refactor (this feature is non-negligible)
